### ISSUE TITLE
[f] Implement knowledge distillation callback (1)

### ISSUE THREAD

**Issue by sieu-n** [krenerd/awesome-modular-pytorch-lightning#23]
- KL-loss based KD
  - self-distillation
  - distillation-based loss functions - 
  - Early stopping KD - On the Efficacy of Knowledge Distillation
- FitNets
  - A Comprehensive Overhaul of Feature Distillation

Can this be implemented as some sort of `pytorch_lightning.callbacks.Callback` ?
